BaragonCore/src/main/java/com/hubspot/baragon/models/BaragonRequest.java

```diff
@@ -27,15 +27,19 @@
   @NotNull
   private final List<UpstreamInfo> removeUpstreams;
 
+  private Optional<Boolean> allowBasepathOverride = Optional.of(false);
+
   @JsonCreator
   public BaragonRequest(@JsonProperty("loadBalancerRequestId") String loadBalancerRequestId,
                         @JsonProperty("loadBalancerService") BaragonService loadBalancerService,
                         @JsonProperty("addUpstreams") List<UpstreamInfo> addUpstreams,
-                        @JsonProperty("removeUpstreams") List<UpstreamInfo> removeUpstreams) {
+                        @JsonProperty("removeUpstreams") List<UpstreamInfo> removeUpstreams,
+                        @JsonProperty("allowBasepathOverride") Optional<Boolean> allowBasepathOverride) {
```

Comment:
i'm wary to go down this route because we're not explicit about what this will override... i'd prefer it we had a new optional property `replaceServiceIds`, which will be removed as part of the Request. one question to answer is should we fail the request if it mentions a service that doesn't exist (probably not)